arch/cpu/nrf/dev/usb-arch.c

```diff
@@ -76,8 +77,8 @@ power_event_handler(nrfx_power_usb_evt_t event)
 void
 usb_arch_init(void)
 {
-  const uint16_t serial_num_high_bytes = (uint16_t)NRF_FICR->DEVICEADDR[1] | 0xC000; // The masking makes the address match the Random Static BLE address.
-  const uint32_t serial_num_low_bytes  = NRF_FICR->DEVICEADDR[0];
+  const uint16_t serial_num_high_bytes = nrf_ficr_deviceid_get(NRF_FICR, 1) | 0xC000;
```

Comment:
Note DEVICEID and DEVICEADDR are different registers in FICR (https://infocenter.nordicsemi.com/pdf/nRF52840_PS_v1.1.pdf#%5B%7B%22num%22%3A160%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C85.039%2C774.623%2Cnull%5D).

There doesn't seem to be an nrfx method to read DEVICEADDR.

Reply:
Yes, but `DEVICEADDR` only exists in in boards with radio, so the application core does not have it, while `DEVICEID` exists in all boards. `DEVICEID` is also an unique identifier according to the nrf5340 data sheet (https://infocenter.nordicsemi.com/pdf/nRF5340_OPS_v0.5.pdf). Is there a reason to specifically use `DEVICEADDR`? I know that it is used in the SDK but the SDK does not support nrf5340.

Reply:
Good point about non-radio cores!

On the nRF52840, if I recall correctly, the DFU bootloader will generate the SN from DEVICEADDR. The matching serial numbers are used by nrfutil to find the bootloader port after a DFU reset.

Reply:
I'll do some research on this. I believe nordic came up with something for this. :)

Reply:
It seems that you are correct, based on the nrf52840 port I got the following:

```
Bus 020 Device 020: ID 1915:520f Nordic Semiconductor ASA nRF52 USB CDC on Contiki-NG  Serial: E46F5CD692A5
```

```
Bus 020 Device 021: ID 1915:521f Nordic Semiconductor ASA Open DFU Bootloader  Serial: E46F5CD692A5
```

and we know that the nRF5 SDK uses the `DEVICEADDR` to generate the serial.

May I suggest to leave it using the `DEVICEID` until we add DFU support? 
I'm curious on how nordic intends to handle this for the nrf5340. 
I'd say this is a design mistake on their side to make the USB serial depend on the `DEVICEADDR` which is available on boards with radio.